### PR TITLE
fix: improve next URL on OAuth

### PR DESCRIPTION
**Pull request by dpgaspar**
### Description

Improves logic on next URL OAuth. Adds type annotations

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Is CRUD MVC related.
- [ ] Is Auth, RBAC security related.
- [ ] Changes the security db schema.
- [ ] Introduces new feature
- [ ] Removes existing feature

### CODE REVIEW

**Comment by dpgaspar** [comment]
@ashb if you have time a review would be nice

File: flask_appbuilder/security/views.py [hunk 1]
```diff
@@ -735,11 +706,15 @@ def oauth_authorized(self, provider):
...
-            try:
-                next_url = state["next"][0] or self.appbuilder.get_url_for_index
-            except (KeyError, IndexError):
-                next_url = self.appbuilder.get_url_for_index
-
+            next_url = self.appbuilder.get_url_for_index
+            # Check if there is a next url on state
+            if "next" in state and len(state["next"]) > 0:
+                parsed_uri = urlparse(state["next"][0])
+                if parsed_uri.netloc != request.host:
```
Comment (ashb) [inline flask_appbuilder/security/views.py:713]: Is it possible to check the "base url" instead? I can't remember if that information makes it down to FAB or not (I suspect not?) but if it does checking that would be better.

Also should we check port number too? (which I guess could be optional/empty on either side?)

File: flask_appbuilder/security/views.py [hunk 2]
```diff
@@ -735,11 +706,15 @@ def oauth_authorized(self, provider):
...
+                    log.warning("Got an invalid next URL: %s", parsed_uri.netloc)
+                    next_url = self.appbuilder.get_url_for_index
```
Comment (ashb) [inline flask_appbuilder/security/views.py:715]: Already set outside the check.

```suggestion
```

File: flask_appbuilder/security/views.py (hunk 1 above)
Comment (ashb) [reply flask_appbuilder/security/views.py:713]: Oh `request.host` already includes non-standard ports

Comment (dpgaspar) [reply flask_appbuilder/security/views.py:713]: yes, so I think it's fine, do you agree?

Comment (ashb) [reply flask_appbuilder/security/views.py:713]: Yeah request.host is good.


Is it worth checking `parsed_uri.path` starts with https://flask.palletsprojects.com/en/2.0.x/api/?highlight=request#flask.Request.root_path for  too?

**Review by ashb — APPROVED** [review]
This is a good improvement, I'll let you decide if it's also worth checking `root_path`

**Pull request merged**